app/lib/common/widgets/scrollable_list_tab_scroller.dart

```diff
@@ -363,37 +365,51 @@ class DefaultHeaderWidget extends StatefulWidget {
 }
 
 class _DefaultHeaderWidgetState extends State<DefaultHeaderWidget>
-    with SingleTickerProviderStateMixin {
-  late final TabController _tabController;
+    with TickerProviderStateMixin {
+  TabController? _tabController;
 
   @override
   void initState() {
     super.initState();
+    _setController();
+  }
+
+  void _setController() {
+    _tabController?.dispose();
```

Comment:
we are doing `_setController` in two places: first at init but _also_ if the widget count changed. That is the actual fix. Because we need to reset the controller if the count has changed and for that we need to _discard_ the old controller and remove the previous listener or it will cause problems being out of sync. In that didWidgetUpdate the dispose doesn't take place and as we have no way of figuring out if a `late` was initialized, having the tabController disposed _if_ it is existing is what we want to do  here.

Reply:
Hmm, I understand the implication but I'd probably look for some other safe approach. Fine for now if you're confident.